osmomath/decimal.go

```diff
@@ -338,10 +350,36 @@ func (d BigDec) MulMut(d2 BigDec) BigDec {
 	return BigDec{d.i}
 }
 
+func (d BigDec) MulDec(d2 Dec) BigDec {
+	copy := d.Clone()
+	copy.MulDecMut(d2)
+	return copy
+}
+
+func (d BigDec) MulDecMut(d2 Dec) BigDec {
+	d.i.Mul(d.i, d2.BigIntMut())
+	d.i = chopPrecisionAndRoundSdkDec(d.i)
+
+	if d.i.BitLen() > maxDecBitLen {
+		panic("Int overflow")
+	}
```

Comment:
nit: this logic gets reused 5/6 times in this PR, maybe could be a single line helper.

Reply:
Lets do that in a second PR, agreed tho